### DISCUSSION TITLE
CogAlG Understanding Review

### DISCUSSION THREAD

**Discussion by Tapas15** [boris-kz/CogAlg#72] [General]
Hi Chris and Boris,
I want to thank you for giving me the opportunity and time during my busy schedule, I am very happy to express my gratitude give upon my views what I actually know or want to know. it is we going to work like some sort of human understanding ability that kind of thing or algorithm which is going to work correct me if am I rite. I m trying to understand the code in a very simple manner are we looking into some kind of pattern, the codes which run correctly without giving any error  line_patterns.py, line_patterns_odd.py, line_patterns_olp.py, frequency_seperation_audio.py. al are running correctly. and final class_cluster.py is running with an error. 
Without not going into much detail about the code my question is that what we are trying to achieve?

**Comment by boris-kz** [comment]
Thanks Tapas! The purpose of this algorithm is general unsupervised learning, which I think is the same as pattern discovery.
I tried to explain it in Readme, not sure what parts are not clear?

**Reply by Risingabhi** [reply]
HI This is Abhishek v from TDI. What i could understand from readme , is that patterns are the way we make sense of things, In Computer Vision, typically our biggest asset is pixels, we join many together to kind of do Feature Extraction, and hence make some reasonable inference about detection.
Now, a human mind doesnt work that way, we have learned the patterns and we can quickly locate the object of interest in a live scene, or pic instantly( think of your REFLEX actions) , this means that CV algorithm need to graduate to a higher level of operations to perform tasks - detection while keeping various real life situations in consideration for eg rotation of item should not necessarily mean we need to add more images in training set, we are currently using as many cases to replicate real life as possible, however, CV algorithms fail in identifying same object in different light conditions, rotation, stretching etc, idea is to make a UNSUPERVISED algo that learns PATTERNS and makes right conclusion even when algo has not seen same instance earlier.
These are my understanding, @boris-kz correct me if i am wrong

**Reply by boris-kz** [reply]
Hi Abshishek. Human vision also starts from "pixels", represented by light-detecting rods and cones in retina. And it also discovers patterns through the hierarchy of visual cortex, vaguely similar to the hierarchy of layers in DL. So, learning always builds on previous observations, there is no way around it.
But the way this is done is very different in my approach, please see "Comparison to ANN and BNN" section in readme.
